### PR TITLE
query extensions

### PR DESCRIPTION
**Pull request by Garvshah01**

### CODE REVIEW

File: app/models/product.rb [hunk 1]
```diff
@@ -26,6 +26,10 @@ class Product < ApplicationRecord
   before_validation :set_name_default
   before_validation :set_discount_price
 
+  scope :enabled_products, -> { where enabled: true }
+  scope :in_line_items, -> { joins(:line_items).distinct }
+  scope :title_in_line_items, -> { in_line_items.pluck(:title) }
```
Comment (shivamvinsol) [inline app/models/product.rb]: this is NOT a scope. 
Scopes are chainable and should return activerecord relation object (AREL)

Add these as class methods

File: app/models/order.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@ class Order < ApplicationRecord
 
   validates :name, :address, :email, presence: true
 
+  scope :by_date, Proc.new {|from, to| where created_at: (from||Date.today)..(to||Date.today) }
```
Comment (shivamvinsol) [inline app/models/order.rb]: - spacing issues 🤦🏻 
- Why are we using `Proc.new` format here?
- Can we use default parameters?
- whats the datatype of `created_at`? What datatype should the default parameters be?
- `user.orders.by_date​ ​should​ ​work​ ​and​ ​it​ ​should​ ​return​ ​orders​ ​placed​ ​by​ ​user
between​ ​given​ ​dates.​ ​Check​ ​the​ ​query​ ​difference​ ​in​ ​this​ ​one​ ​and​ ​the​ ​above​ ​one`
  - Add a text file where you run the queries, and explain the difference

**Review by shivamvinsol — CHANGES_REQUESTED** [review]

File: app/models/order.rb (hunk 2 above)
Comment (Garvshah01) [reply app/models/order.rb]: [order_by_date.txt](https://github.com/Garvshah01/Depot/files/11280133/order_by_date.txt)

File: app/models/product.rb (hunk 1 above)
Comment (Garvshah01) [reply app/models/product.rb]: done

File: app/models/order.rb (hunk 2 above)
Comment (Garvshah01) [reply app/models/order.rb]: done

File: app/models/order.rb [hunk 3]
```diff
@@ -6,6 +6,8 @@ class Order < ApplicationRecord
 
   validates :name, :address, :email, presence: true
 
+  scope :by_date, ->(from: DateTime.parse(Date.today.to_s), to: DateTime.now) { where created_at: from..to }
```
Comment (shivamvinsol) [inline app/models/order.rb]: read about current vs now

Comment (shivamvinsol) [reply app/models/order.rb]: read datetime functions

Comment (Garvshah01) [reply app/models/order.rb]: done

File: app/models/order.rb [hunk 4]
```diff
@@ -6,6 +6,8 @@ class Order < ApplicationRecord
 
   validates :name, :address, :email, presence: true
 
+  scope :by_date, -> (from: DateTime.now.at_beginning_of_day, to: DateTime.now) { where created_at: from..to }
```
Comment (shivamvinsol) [inline app/models/order.rb]: read diff about now vs current
why are we not doing end_of_day in to?
use beginning_of_day

Comment (Garvshah01) [reply app/models/order.rb]: done

**Review by shivamvinsol — APPROVED** [review]

**Pull request merged**